engine/sdk_tx_getter.go

```diff
@@ -0,0 +1,51 @@
+package engine
+
+import (
+	"context"
+	"slices"
+	"time"
+
+	sdk "github.com/bitcoin-sv/go-sdk/transaction"
+	"github.com/bitcoin-sv/spv-wallet/engine/spverrors"
+	"iter"
+)
+
+type sdkTxGetter struct {
+	client *Client
+}
+
+func newSDKTxGetter(client *Client) *sdkTxGetter {
+	return &sdkTxGetter{client: client}
+}
+
+func (g *sdkTxGetter) GetTransactions(ctx context.Context, ids iter.Seq[string]) ([]*sdk.Transaction, error) {
+	db := g.client.Datastore().DB()
+
+	queryIDsCtx, cancel := context.WithTimeout(ctx, 20*time.Second)
+	defer cancel()
+
+	var hexes []struct {
+		Hex string
+	}
+
+	err := db.
+		WithContext(queryIDsCtx).
+		Model(&Transaction{}).
+		Where("id IN (?)", slices.Collect(ids)).
```

Comment:
how big the collection of ids you think it will be ? asking to avoid passing large number of IDs to the IN

Reply:
Not big. Usually several; sometimes several dozen.

I think it is already a boost, because in previous implementation there was `id = ? or id = ? or ...`. 
And also, previously, the whole object (all columns) was selected. Now only `hex`es are obtained (`SELECT hex from ...`)

Reply:
ok